### DISCUSSION TITLE
Unable to download & install meshcentral agents

### DISCUSSION THREAD

**Discussion by bilal-rabbani** [Ylianst/MeshCentral#6270] [Q&A] [answered]
Greetings,
i'm new to meshcentral, 
i;m having few issues

1. how can i know that meshcentral is connected to the domain name i provided, i got domain name from [no-IP](noip.com)
2. i'm unable to download meshcentral agent, i tried downloaded meshcentral agent many time but was able to download twice only, but i'm unable to install agents to other computer,  the issue says there is problem installing the program
 my antivirus & firewall both are off, i have searched for solution online & i think my ports are blocking the connection but in firewall settings i have allowed meshcentral through port 80 & 443
![Screenshot 2024-07-23 141840](https://github.com/user-attachments/assets/7f7aeeaf-f84f-478f-93ce-7bfcc9ba81a6)
![Screenshot 2024-07-23 141916](https://github.com/user-attachments/assets/e904905b-5a85-41c8-a4c7-f2aef6227086)

**Comment by si458** [comment]
Can u share ur config.json?
as u seem to be like the 4th people with this issue, yet we can't replicate it?

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 144246](https://github.com/user-attachments/assets/13379a25-1f74-4eb9-bdf2-f81f65a16b1e)

**Reply by si458** [reply]
You have both wanonly and lanonly set as true, you should only use one or other or neither.
And you should be access ur Web ui using the dns name u set in the cert value

**Reply by bilal-rabbani** [reply]
got it 
![Screenshot 2024-07-23 151643](https://github.com/user-attachments/assets/0eeaec69-7bb0-47ab-b000-50987d8603d3)

**Comment by si458** [comment]
If u have set the `cert` value to a dns name,
you should be accessing ur Web ui using that as the url and not localhost

**Reply by bilal-rabbani** [reply]
exactly, but as i put dns in browser it's always an error sometimes it says "400 bad requests" sometimes it says "time out" or something like that.

**Reply by si458** [reply]
You need to check ur router is port forwarding 80 and 443 to ur meshcentral machine, and also, depending on ur router, you need to set up hairpin nat (this is mainly mikrotik routers tho)

**Reply by bilal-rabbani** [reply]
i'm sorry but i don't know how to do that,

**Reply by si458** [reply]
> i'm sorry but i don't know how to do that,  

Then sadly meshcentral won't work correctly as its designed to be run on the Internet on a server (in the cloud hehe)
Or
If u want to run it at home, u need to port forward 80 and 443 in my router to your meshcentral router

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 152452](https://github.com/user-attachments/assets/5d5cc75f-a444-44d3-9beb-7059eba81dd8)
can it be done from here?
can you kindly suggest any youtube video or any other platform to resolve this issue?

**Reply by bilal-rabbani** [reply]
can it be done from here?
![Screenshot 2024-07-23 152817](https://github.com/user-attachments/assets/2ed58698-5a63-4ea6-9e7b-d78b54a7a15a)

**Reply by si458** [reply]
> ![Screenshot 2024-07-23 152452](https://github.com/user-attachments/assets/5d5cc75f-a444-44d3-9beb-7059eba81dd8)
> can it be done from here?
> can you kindly suggest any youtube video or any other platform to resolve this issue?
> 

Yes, that's the page, so externally 80 to internally 80 and 443 externally to 443 internally and make sure to set ur meshcentral ip address as the internal ip address

**Reply by bilal-rabbani** [reply]
where can i get my meshcentral ip? sorry for asking such basic questions

**Reply by si458** [reply]
> where can i get my meshcentral ip? sorry for asking such basic questions

It's just the ip address of the machine that's running meshcentral

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 153820](https://github.com/user-attachments/assets/01694b59-c68c-4d8b-bf8d-05f79ae3a8bd)
is this the meshcentral ip?

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 154012](https://github.com/user-attachments/assets/b6fc2747-8bc3-4c99-a7be-6ab68bd1ca1f)
is the setting ok? except ip address,

**Reply by si458** [reply]
> ![Screenshot 2024-07-23 153820](https://github.com/user-attachments/assets/01694b59-c68c-4d8b-bf8d-05f79ae3a8bd)
> is this the meshcentral ip?

No, that's ur external ip address.
The internal ip address is the ip address of ur computer.
U can check that in the windows settings panel
https://www.malwarebytes.com/cybersecurity/computer/how-to-find-your-ip-address-on-windows-10

**Reply by si458** [reply]
> ![Screenshot 2024-07-23 154012](https://github.com/user-attachments/assets/b6fc2747-8bc3-4c99-a7be-6ab68bd1ca1f)
> is the setting ok? except ip address, 

No, u have one port per line
So 80 80 80 80
Then next line 
443 443 443 443

**Reply by si458** [reply]
Also, that screenshot shows u are using a mobile 3g/4g dongle, so that won't work!

Because your dongle is being double natted by the mobile carriers so it won't work!

You would need to look into something like cloudflare tunnel or ngrok both free but not free in a sense

**Reply by bilal-rabbani** [reply]
is this setting ok? moreover do i have to specify any service? or change wan interface 
![Screenshot 2024-07-23 154755](https://github.com/user-attachments/assets/8dd22dec-6d9c-492a-a74c-fc1cae7bbca0)

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 155713](https://github.com/user-attachments/assets/ab258496-25d3-4c13-9a31-a6c42179f9e5)
couldn't find cloudflare tunnel or ngrok, am i doing something wrong ?

**Reply by si458** [reply]
> is this setting ok? moreover do i have to specify any service? or change wan interface ![Screenshot 2024-07-23 154755](https://private-user-images.githubusercontent.com/139207000/351295287-8dd22dec-6d9c-492a-a74c-fc1cae7bbca0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MzMyMzgsIm5iZiI6MTcyMTczMjkzOCwicGF0aCI6Ii8xMzkyMDcwMDAvMzUxMjk1Mjg3LThkZDIyZGVjLTZkOWMtNDkyYS1hNzRjLWZjMWNhZTdiYmNhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QxMTA4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNjVlNzE2OWRlYmUyZjRiNGI0YTZjZTY5OTY3YTY0MWM0NzU3OTY0ZDM4MmM2MzVhNjk5MDU4ZWRmNGQxY2I1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TKTqOnroDKOE26Idku3z3u2nCWC8Cf2qS6qjFp8NZy8)

that all looks correct, just make sure the WAN interface is set to whichever interface is your internet connection!
as i can see in the 2nd screenshot i sent a few like pppoe interfaces

just ignore me about `cloudflare tunnel or ngrok` i was saying that IF your internet connection is using a 3G/4G dongle then port forwarding wont work because of double nat, so you will need to look into using a service like cloudflare tunnel or ngrok

**Reply by bilal-rabbani** [reply]
how can i know which is my internet connection interface ? &  thanks for  your  time bro you helped alot

**Reply by si458** [reply]
i dont no, try checking the WAN service page, ive never seen that UI before so i dont no what router you are using

**Reply by bilal-rabbani** [reply]
![Screenshot 2024-07-23 165923](https://github.com/user-attachments/assets/2fcedae1-1a6e-404e-b171-f4ee97052130)
![Screenshot 2024-07-23 170118](https://github.com/user-attachments/assets/02962736-4130-4a39-82a2-1329b07576cb)
bro got any idea?

**Reply by si458** [reply]
not a clue sorry, but a google reveals this youtube video tho which might help? (say the logo on last screenshot)
https://www.youtube.com/watch?v=-p45Iw2Hecw

**Reply by bilal-rabbani** [reply]
thank you so much now i can  access my Web ui using the dns name but the problem presists & the problem is that i can only download meshcentral agent once if i want to download it again i have to restart the server, & when i try to run that agent in other pc it says meshcentral background service agent has stopped working, do you know what is this issue?

**Reply by si458** [reply]
strange one, change `minify` to `false` inside your config.json, then restart the meshcentral service,
then close browser, reopen then try again?
the was a bug just discovered with minify

**Reply by si458** [reply]
also one thing to try is go into the `My Server` tab on left hand side, then click `Show server error log` and see if the is any errors listed there

**Reply by bilal-rabbani** [reply]
i have changed minify to false but nothing changed the problem persists while downloading agent error says check internet connection
![Screenshot 2024-07-23 141916](https://github.com/user-attachments/assets/083516b1-5033-4094-8534-55ab3af95334)

**Reply by si458** [reply]
you still arent accessing the website correctly...
the url in ur pic shows `https://localhost` you should be access it via the DNS name u set in `cert`

**Reply by bilal-rabbani** [reply]
> also one thing to try is go into the `My Server` tab on left hand side, then click `Show server error log` and see if the is any errors listed there

it says "server has no error logs"

**Reply by bilal-rabbani** [reply]
> you still arent accessing the website correctly... the url in ur pic shows `https://localhost` you should be access it via the DNS name u set in `cert`

this picture is from before rn i'm accessing it through dns name

**Reply by si458** [reply]
have you tried a different browser or a different computer?

**Reply by bilal-rabbani** [reply]
> have you tried a different browser or a different computer?

yeah i have tried but same issue my antivirus & firewall both are off

**Reply by si458** [reply]
how did you install meshcentral? did you use the meshcentralinstaller or the `npm install` ?

**Reply by bilal-rabbani** [reply]
> how did you install meshcentral? did you use the meshcentralinstaller or the `npm install` ?

Npm install

**Reply by si458** [reply]
the only thing to check is the `agents` folder and make sure its full of exes
so `YOURROOTFOLDER/node_modules/meshcentral/agents/`

**Reply by bilal-rabbani** [reply]
> the only thing to check is the `agents` folder and make sure its full of exes so `YOURROOTFOLDER/node_modules/meshcentral/agents/`

![Screenshot 2024-07-23 221552](https://github.com/user-attachments/assets/3b4d261f-4054-405b-912d-6f497735aaf3)

**Reply by si458** [reply]
im really not sure im afraid, only thing left you can try is
1. simply try restarting your computer
2. running meshcentral in debug mode
`node node_modules/meshcentral --debug`
3. watch console output and see if you get any errors/warnings

**Reply by bilal-rabbani** [reply]
> im really not sure im afraid, only thing left you can try is
> 
> 1. simply try restarting your computer
> 2. running meshcentral in debug mode
>    `node node_modules/meshcentral --debug`
> 3. watch console output and see if you get any errors/warnings

![Screenshot 2024-07-23 231331](https://github.com/user-attachments/assets/0ac11587-2dab-411b-9381-905d66703b94)

**Reply by si458** [reply]
ok and do you get any messages/logs etc there when you access the website using ur dns name and try downloading the agent?

**Reply by bilal-rabbani** [reply]
> ok and do you get any messages/logs etc there when you access the website using ur dns name and try downloading the agent?

all it says is "check internet connection" kindly refer to the first & 2nd screenshot

**Reply by si458** [reply]
> > ok and do you get any messages/logs etc there when you access the website using ur dns name and try downloading the agent?
> 
> all it says is "check internet connection" kindly refer to the first & 2nd screenshot

i meant, is the any extra outputs in the black window where u ran the debug command?

**Reply by bilal-rabbani** [reply]
> ok and do you get any messages/logs etc there when you access the website using ur dns name and try downloading the agent?

are last two files normal?
![Screenshot 2024-07-23 234644](https://github.com/user-attachments/assets/8cb37e4c-19ed-4903-9c07-604890f5d9f1)

**Reply by bilal-rabbani** [reply]
> > > ok and do you get any messages/logs etc there when you access the website using ur dns name and try downloading the agent?
> > 
> > 
> > all it says is "check internet connection" kindly refer to the first & 2nd screenshot
> 
> i meant, is the any extra outputs in the black window where u ran the debug command?

i didn;t get any log or message

**Reply by si458** [reply]
sadly i have no more ideas im afraid,
only thing i can think of is you have some internet issues when it comes to downloading the exes,
maybe the browser is blocking the files being downloaded

also yes the package.json and package-lock.json are normal, they should be there

**Comment by bilal-rabbani** [comment, marked as answer]
> Greetings, i'm new to meshcentral, i;m having few issues
> 
> 1. how can i know that meshcentral is connected to the domain name i provided, i got domain name from [no-IP](noip.com)
> 2. i'm unable to download meshcentral agent, i tried downloaded meshcentral agent many time but was able to download twice only, but i'm unable to install agents to other computer,  the issue says there is problem installing the program
>    my antivirus & firewall both are off, i have searched for solution online & i think my ports are blocking the connection but in firewall settings i have allowed meshcentral through port 80 & 443
>    ![Screenshot 2024-07-23 141840](https://private-user-images.githubusercontent.com/139207000/351279462-7f7aeeaf-f84f-478f-93ce-7bfcc9ba81a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MTM4OTgsIm5iZiI6MTcyMTgxMzU5OCwicGF0aCI6Ii8xMzkyMDcwMDAvMzUxMjc5NDYyLTdmN2FlZWFmLWY4NGYtNDc4Zi05M2NlLTdiZmNjOWJhODFhNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNFQwOTMzMThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NmIwMGM1OWRhM2E2ODlhZWRiZTdjOWM5N2I3YzQ1NDU2MDI0MzNjMDQ1OTM3ZTlhNzM0NGRhYTBiMmUzNmJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ykq67q4ZXbTKp1_0v6tRXuhYX9SphbIio2Re-5QqZUY)
>    ![Screenshot 2024-07-23 141916](https://private-user-images.githubusercontent.com/139207000/351279555-e904905b-5a85-41c8-a4c7-f2aef6227086.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MTM4OTgsIm5iZiI6MTcyMTgxMzU5OCwicGF0aCI6Ii8xMzkyMDcwMDAvMzUxMjc5NTU1LWU5MDQ5MDViLTVhODUtNDFjOC1hNGM3LWYyYWVmNjIyNzA4Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNFQwOTMzMThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Yzg0ODkwMmI1NmM5YzVjNjdmZmJmODBkYTJiMzk0YTIzNjhhMGM3NWE2YTRmODZjZmQxY2Y2NjEyYmQyNTBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.b_0V6CKCq846zR-lYYLfxbjzdmcDCtJUAXN5oRzvuPk)

i have finally figured  the solution, the solution is to turn off extensions(specially idm), i have disabled extentions & i'm able to download agents as many times i want. thank you so much @si458 for your efforts.

**Reply by si458** [reply]
glad its sorted! 👍
